bonfire/utils.py

```diff
@@ -143,6 +145,19 @@ def _gl_certfile(self):
 
         return cert_fname
 
+    @cached_property
+    def _gh_auth_headers(self):
+        gh_token = os.getenv("GITHUB_TOKEN")
+        if gh_token:
+            return {"Authorization": f"token {gh_token}"}
+        return None
+
+    @property
+    def _jenkins_or_local(self):
```

Comment:
I see what you are going for, IMO this might be cleaner if we just have:
```python
GH_API_URL = os.getenv('GITHUB_API_URL', "https://api.github.com")
GH_BRANCH_URL = GH_API_URL.rstrip('/') + "/repos/{org}/{repo}/git/refs/heads/{branch}"
```

Then we will just configure the app-sre Jenkins jobs to use:
GITHUB_API_URL
GITHUB_TOKEN

and then the bonfire code doesn't need to have the concept of "on jenkins" or "not on jenkins" baked into its code.